nixos/modules/tasks/network-interfaces.nix

```
@@ -541,6 +544,28 @@ in
       '';
     };
 
+    networking.machineId = mkOption {
+      default = null;
+      example = "b23d9f80c4494c00b632f2f6448aaa72";
+      type = types.nullOr (types.strMatching "[[:xdigit:]]{32}");
+      description = lib.mdDoc ''
+        The [128-bit machine ID](https://www.freedesktop.org/software/systemd/man/latest/machine-id.html)
+        of the machine, formatted as 32 hexadecimal characters.
+
+        You should try to make this ID unique among your machines.
+        The ID is generated randomly on boot, if it does not already exists.
+
+        The current machine ID can be found in `/etc/machine-id`.
+
+        The primary use case is to ensure a unique identifier for the computer, used
+        by e.g. systemd journal.
+        If `networking.machineId` is set, but `networking.hostId` is not set, the
+        hostId will be derived from the machineId specified.
```

This word seems redundant to me:
```suggestion
        hostId will be derived from the machineId.
```